### PR TITLE
fix(umd): should ensure exectution if entry module is wrapped

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2841** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2841?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2839** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2839?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2839?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2838** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2838?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown/src/ecmascript/format/umd.rs [hunk 1]
```diff
@@ -127,6 +128,38 @@ pub fn render_umd<'code>(
...
+        // require_xxx();
+        // FIXME: should set the umd's exports with `require_xxx()`
+        source_joiner.append_source(concat_string!(wrapper_ref_name, "();\n"));
+      }
+      WrapKind::None => {}
+    }
+  }
+
   //  exports
   if let Some(exports) = render_chunk_exports(ctx, Some(&export_mode)) {
```
Comment (hyf0) [inline crates/rolldown/src/ecmascript/format/umd.rs:164]: This should be refactored with considering module being wrapped.

@underfin do you have time for this?

Comment (hyf0) [reply crates/rolldown/src/ecmascript/format/umd.rs:164]: iife format should be refactored too I guess.

Comment (underfin) [reply crates/rolldown/src/ecmascript/format/umd.rs:164]: The logic of `module being wrapped` moving into `render_chunk_exports`?

**Review by underfin — APPROVED** [review]

File: crates/rolldown/src/ecmascript/format/umd.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/src/ecmascript/format/umd.rs:164]: I guess so. Need to render different `module.exports = require_xxx()`/`return require_xxx()` based on output format.

Comment (underfin) [reply crates/rolldown/src/ecmascript/format/umd.rs:164]: Yeah, you are right, but it needs to low priority for me at now, let us delay to do it.

**Comment by hyf0** [comment]
### Merge activity

* **Nov 21, 5:02 AM EST**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2839) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**